src/components/shared/Terminal.tsx

```diff
@@ -191,10 +322,39 @@ function Terminal(prop: {
   const handleSubmit = (e: React.FormEvent) => {
     e.preventDefault();
     const result = isValid(input);
-    setCommands([...commands, input, ...result.err, ...result.out]);
+    const newHistory = [...inputHistory, input];
+    if (result.err.length > 0 && result.err[0] === 'CLEAR') {
```

Comment:
'CLEAR' should be a constant to avoid people from making typos regarding case

Reply:
Agreed, will add 'CLEAR' to a constants file (backlog task)